src/main/scala/cromwell/engine/workflow/SingleWorkflowRunnerActor.scala

```diff
@@ -79,18 +85,19 @@ case class SingleWorkflowRunnerActor(source: WorkflowSourceFiles,
       log.info(s"$tag: workflow ID UUID($id)")
       workflowManager ! SubscribeToWorkflow(id)
 
-      if (getAbortJobsOnTerminate) {
-        Runtime.getRuntime.addShutdownHook(new Thread() {
-          override def run(): Unit = {
-            workflowManager ! WorkflowAbort(id)
-            log.info(s"$tag: Waiting for workflow $id to abort...")
-            while(stateName != Done)
-              Thread.sleep(1000)
-            log.info(s"$tag: Workflow $id aborted.")
-          }
-        })
+      if (abortJobsOnTerminate) {
+        sys.addShutdownHook {
+          workflowManager ! WorkflowAbort(id)
+          log.info(s"$tag: Waiting for workflow $id to abort...")
+          Await.ready(donePromise.future, Duration.Inf)
```

Comment:
@cjllanwarne I'm probably missing a variety of other bits but one thing is that `Await.ready` calls `blocking {}` under the hood, which gives a heads up to your execution context that danger is incoming so it can try to manage things a bit better. You also might get better granularity than in your polling loop.

Another angle which seems trivial but I'd argue it's very much not is that it makes it more clear that blocking is happening. `Await` pops out like a giant red alarm bell, at least to me.

Iff I haven't missed anything key the differences would be relatively minor.

@scottfrazer I believe both `Awaits` really do block the thread, but they should both have that `blocking` keyword in there (nothing prevents someone from doing that themselves) which helps make things more sane. That's part of why they're also viewed with suspicion.